src/backends/perf_pt/collect.c

```diff
@@ -264,8 +265,20 @@ open_perf(pid_t target_tid) {
     // No skid.
     attr.precise_ip = 3;
 
-    // Acquire file descriptor through which to talk to Intel PT.
-    ret = syscall(SYS_perf_event_open, &attr, target_tid, -1, -1, 0);
+    // Acquire file descriptor through which to talk to Intel PT. This syscall
+    // could return EBUSY, meaning another process or thread has locked the
+    // Perf device.
+    struct timespec wait_time = {0, 1000 * 20}; // 20 microseconds.
+    for (int tries = 500; tries > 0; tries--) {
```

Comment:
Can we pull this and the constant above into `#define`s at the top of the file?

Reply:
I did consider this, along with the timeout too. One moment.